src/lib/hooks/use-best-quote/index.ts

```diff
@@ -237,35 +235,33 @@ async function getFlashMintQuote(
     if (flashMintQuote === null) return savedQuote
     // For redeeming return quote immdediately
     if (!isMinting) return flashMintQuote
-    // As a safety measure we're aborting after 30 seconds take whatever quote we got
-    const now: number = new Date().getTime()
-    const diffSinceStart = (now - timestamp) / 1000
-    console.log(timestamp, now, diffSinceStart)
-    if (diffSinceStart > 10) return flashMintQuote
-    // For minting check if we got a quote that is lower/equal than the input token amount
-    // - since we should never go above what the user entered intitially. Additionally,
-    // we're checking if there might be a too big of a difference to the original input amount.
-    const { diff, shouldReturn } = shouldReturnQuote(
-      inputTokenAmountWei.toBigInt(),
-      flashMintQuote.inputTokenAmount.toBigInt()
-    )
-    if (shouldReturn) return flashMintQuote
-    // Save last fetched quote to be able to return something if next run fails
     savedQuote = flashMintQuote
-    console.log('diff:', diff.toString())
-    const buffer = 1
-    console.log(diff < 0, 100 - Math.floor(Math.abs(diff)) - buffer)
-    console.log(diff >= 0, 100 + Math.round(Math.abs(diff)) - buffer)
-    if (diff < 0) {
-      // The quote input amount is too high, reduce
-      indexTokenAmount =
-        (indexTokenAmount * BigInt(100 - Math.floor(Math.abs(diff)) - buffer)) /
-        BigInt(100)
-    } else {
-      // The quote input amount is too low from the original input, increase
-      indexTokenAmount =
-        (indexTokenAmount * BigInt(100 + Math.round(Math.abs(diff)) - buffer)) /
-        BigInt(100)
+
+    console.log('estimated index token amount', indexTokenAmount.toString())
+    const diff = inputTokenAmountWei
+      .sub(flashMintQuote.inputTokenAmount)
+      .toBigInt()
+    console.log('diff', diff.toString())
+    const factor = determineFactor(diff, inputTokenAmountWei.toBigInt())
+    console.log('factor', factor.toString())
+
+    indexTokenAmount = (indexTokenAmount * factor) / BigInt(10000)
+    console.log('new index token amount', indexTokenAmount.toString(), t)
+
+    if (diff < 0 && t === 1) {
+      t++ // loop one more time to stay under the input amount
     }
   }
+
+  return savedQuote
+}
+
+const determineFactor = (diff: bigint, inputTokenAmount: bigint): bigint => {
+  let ratio = Number(diff.toString()) / Number(inputTokenAmount.toString())
```

Comment:
thoughts on changing this to the below line (reduce the chance of `inputTokenAmount` causing an int overflow for a really big number):

`let ratio = Number(diff / inputTokenAmount)`

Reply:
Good point but I think this would cause rounding errors. It should be unlikely for `inputTokenAmount` to be much higher than 10 mio I'd say.